oemof/outputlib/processing.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+"""
+Modules for providing a convenient data structure for solph results.
+
+Information about the possible usage is provided within the examples.
+"""
+
+import pandas as pd
+from itertools import groupby
+from oemof.network import Node
+from pyomo.core.base.var import Var
+
+
+def get_tuple(x):
+    """
+    Get oemof tuple within iterable or create it.
+
+    Tuples from Pyomo are of type `(n, n, int)`, `(n, n)` and `(n, int)`.
+    For single nodes `n` a tuple with one object `(n,)` is created.
+    """
+    for i in x:
+        if isinstance(i, tuple):
+            return i
+        elif issubclass(type(i), Node):
+            return i,
+        else:
+            pass
+
+
+def get_timestep(x):
+    """
+    Get the timestep from oemof tuples.
+
+    The timestep from tuples `(n, n, int)`, `(n, n)`, `(n, int)` and (n,)
+    is fetched as the last element. For time-independent data (scalars)
+    zero ist returned.
+    """
+    if all(issubclass(type(n), Node) for n in x):
+        return 0
+    else:
+        return x[-1]
+
+
+def remove_timestep(x):
+    """
+    Remove the timestep from oemof tuples.
+
+    The timestep is removed from tuples of type `(n, n, int)` and `(n, int)`.
+    """
+    if all(issubclass(type(n), Node) for n in x):
+        return x
+    else:
+        return x[:-1]
+
+
+def create_dataframe(om):
+    """
+    Create a result dataframe with all optimization data.
+
+    Results from Pyomo are written into pandas DataFrame where separate columns
+    are created for the variable index e.g. for tuples of the flows and
+    components or the timesteps.
+    """
+    # get all pyomo variables including their block
+    block_vars = []
+    for bv in om.component_data_objects(Var):
+        block_vars.append(bv.parent_component())
+    block_vars = list(set(block_vars))
+
+    # write them into a dict with tuples as keys
+    var_dict = {(str(bv).split('.')[0], str(bv).split('.')[-1], i): bv[i].value
+                for bv in block_vars for i in getattr(bv, '_index')}
+
+    # use this to create a pandas dataframe
+    df = pd.DataFrame(list(var_dict.items()), columns=['pyomo_tuple', 'value'])
+    df['variable_name'] = df['pyomo_tuple'].str[1]
+
+    # adapt the dataframe by separating tuple data into columns depending
+    # on which dimension the variable/parameter has (scalar/sequence).
+    # columns for the oemof tuple and timestep are created
+    df['oemof_tuple'] = df['pyomo_tuple'].map(get_tuple)
+    df['timestep'] = df['oemof_tuple'].map(get_timestep)
+    df['oemof_tuple'] = df['oemof_tuple'].map(remove_timestep)
+
+    # order the data by oemof tuple and timestep
+    df = df.sort_values(['oemof_tuple', 'timestep'], ascending=[True, True])
+
+    return df
+
+
+def results(es, om):
+    """
+    Create a result dictionary from the result DataFrame.
```

Comment:
Why do you pass the EnergySystem instance? As I understand it the  EnergySystem instance is became an attribute of the OperationalModel.

The following code should work if you need a shortcut. 
```python
def results(om):
    es = om.es
```

Or is it planned to remove the EnergySystem from the OperationalModel?